test/integration/run-bats.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -e
+
+# Wrapper script to run bats tests for various drivers.
+# Usage: ./run-bats.sh [driver] [subtest]
+
+function build_machine() {
+    cd ${MACHINE_ROOT}
+    ./script/build
+    cd -
+}
+
+function cleanup_machines() {
+    echo "Cleaning up machines..."
+    if [[ $(docker-machine ls -q | wc -l) -ne 0 ]]; then
+      machine rm $(machine ls -q)
+    fi
+    if [[ -d "$MACHINE_STORAGE_PATH" ]]; then
+      rm -r $MACHINE_STORAGE_PATH || true
+    fi
+}
+
+function machine() {
+    ${MACHINE_ROOT}/${MACHINE_BIN_NAME} "$@"
+}
+
+function run_bats() {
+    for bats_file in $(find "$1" -name \*.bats); do
+        # BATS returns non-zero to indicate the tests have failed, we shouldn't
+        # neccesarily bail in this case, so that's the reason for the e toggle.
+        set +e
+        bats "$bats_file"
+        if [[ $? -ne 0 ]]; then
+            EXIT_STATUS=1
+        fi
+        set -e
+        cleanup_machines
+    done
+}
+
+PLATFORM=`uname -s | tr '[:upper:]' '[:lower:]'`
+ARCH=`uname -m`
+
+if [ "$ARCH" = "x86_64" ]; then
+    ARCH="amd64"
+else
+    ARCH="386"
+fi
+
+# Set this ourselves in case bats call fails
+EXIT_STATUS=0
+
+# TODO: Should the script bail out if these are set already?
```

Comment:
Interesting -- even if we don't do everything exactly the same I'd like to keep it as consistent as possible, so perhaps I'll rename our variables to be more congruent.